factory/coreComponentsHandler.go

```diff
@@ -123,7 +123,7 @@ func (mcc *managedCoreComponents) CheckSubcomponents() error {
 	if check.IfNil(mcc.syncTimer) {
 		return errors.ErrNilSyncTimer
 	}
-	if check.IfNil(mcc.RoundHandler()) {
+	if check.IfNil(mcc.coreComponents.RoundHandler) {
```

Comment:
please rename field in coreComponents struct from RoundHandler to roundHandler, and use here directly mcc.roundHandler

Reply:
done